packages/xo-server-sdn-controller/src/index.js

```diff
@@ -693,6 +973,136 @@ class SDNController extends EventEmitter {
+    }
+
+    const client = find(
+      this._ovsdbClients,
+      client => client.host.$ref === poolNetwork.starCenter
+    )
+    if (client === undefined) {
+      return
+    }
+
+    const centerNetwork = centerClient.host.$xapi.getObjectByRef(
+      centerPoolNetwork.network
+    )
+    const remoteNetwork = client.host.$xapi.getObjectByRef(poolNetwork.network)
+    try {
+      await centerClient.resetForNetwork(
+        centerNetwork.uuid,
+        centerNetwork.name_label,
+        true,
+        remoteNetwork.uuid
+      )
+    } catch (error) {
+      log.error('Error while clearing pool center from network', {
+        error,
+        network: remoteNetwork.name_label,
+        host: client.host.name_label,
+        remotePool: remoteNetwork.$pool.name_label,
+        uuid: crossPoolNetworkUuid,
+      })
+    }
+  }
+
+  async _resetCrossPoolConnections(poolNetwork, uuid) {
+    const client = find(
+      this._ovsdbClients,
+      client => client.host.$ref === poolNetwork.starCenter
+    )
+    if (client === undefined) {
+      return
+    }
+
+    const network = client.host.$xapi.getObjectByRef(poolNetwork.network)
+    try {
+      await client.resetForNetwork(network.uuid, network.name_label, true)
+    } catch (error) {
+      log.error('Error while resetting cross pool connections', {
+        error,
+        network: network.name_label,
+        host: client.host.name_label,
+        uuid,
+      })
+    }
+  }
+
+  async _connectNetworks(poolNetwork, centerPoolNetwork, uuid) {
+    const client = find(
+      this._ovsdbClients,
+      client => client.host.$ref === poolNetwork.starCenter
+    )
+    if (client === undefined) {
+      return
+    }
+
+    const centerClient = find(
+      this._ovsdbClients,
+      client => client.host.$ref === centerPoolNetwork.starCenter
+    )
+    const centerNetwork = centerClient.host.$xapi.getObjectByRef(
+      centerPoolNetwork.network
+    )
+    const network = client.host.$xapi.getObjectByRef(poolNetwork.network)
+
+    // Use centerNetwork VNI by convention
+    const { encapsulation = 'gre', vni = '0' } = centerNetwork.other_config
+    try {
+      await Promise.all([
+        client.addInterfaceAndPort(
+          network.uuid,
+          network.name_label,
+          centerClient.host.address,
+          encapsulation,
+          vni,
+          centerNetwork.uuid
+        ),
+        centerClient.addInterfaceAndPort(
+          centerNetwork.uuid,
+          centerNetwork.name_label,
+          client.host.address,
+          encapsulation,
+          vni,
+          network.uuid
+        ),
+      ])
+    } catch (error) {
+      log.error('Error while connecting networks', {
+        error,
+        network: network.name_label,
+        host: client.host.name_label,
+        pool: client.host.$pool.name_label,
+        centerHost: centerClient.host.name_label,
+        centerPool: centerClient.host.$pool.name_label,
+        uuid,
+      })
+      return
```

Comment:
Should the calling code be made aware of the error?

Reply:
I think not. Most of (all from my experience) the error is bad timing between the callbacks.
And the method will be recalled a bit later and everything is okay by then.